### PR TITLE
handle corrupted bytecode

### PR DESCRIPTION
**Pull request by aafa**
Adding some more details on why build failed to be processed

### CODE REVIEW

**Comment by pfn** [comment]
I'm tempted to not merge this change. Mostly, bytecode corruption would be a problem caused by tools doing the wrong thing (in this case, javassist/realm) and since these aren't standard and sort of require development anyway, I would expect one to debug it by throwing the message in manually.

File: src/NativeFinder.scala [hunk 1]
```diff
@@ -42,8 +42,12 @@ object NativeFinder {
       case x: ClassVisitor =>
         (classDir ** "*.class" get) foreach { entry =>
           Using.fileInputStream(entry) { in =>
-            val r = new ClassReader(in)
-            r.accept(x, 0)
+              try {
+                val r = new ClassReader(in)
+                r.accept(x, 0)
+              } catch {
+                case e: Exception => throw new IllegalArgumentException(s"NativeFinder processing `$entry` went wrong, bytecode is corrupted")
```
Comment (pfn) [inline src/NativeFinder.scala]: The error should be more along the lines of

`throw new IllegalArgumentException(s"Failed to process bytecode for `$entry`", e)`

**Comment by aafa** [comment]
What was most striking to me is that javasist silently messed up with bytecode and that was discovered latter on in a build process. I will probably look into this again latter.

Probably you right, if someone has been done something to bytecode outside of android-plugin, responsibility is on them. But still I would be more chill to have this additional level of clarity here.

File: src/NativeFinder.scala (hunk 1 above)
Comment (aafa) [inline src/NativeFinder.scala]: Sure, will update if you gonna merge it in.

**Comment by aafa** [comment]
Commit updated.

**Comment by pfn** [comment]
I guess it doesn't hurt, might as well merge

**Pull request merged**